### ISSUE TITLE
imported rollup-plugin-vue/runtime/normalize.js is error

### ISSUE THREAD

**Issue by guaijie** [vuejs/rollup-plugin-vue#303]
### Version
5.1.2

### Reproduction link
[https://jsfiddle.net/chrisvfritz/50wL7mdz/](https://jsfiddle.net/chrisvfritz/50wL7mdz/)

### Steps to reproduce
```
export default {
    input: components,
    output: {
        dir: 'lib/components',
        format: 'esm'
    },
    plugins: [
           vue()
    ]
};
```

![image](https://user-images.githubusercontent.com/30885718/67658334-420af080-f994-11e9-91a2-4b66539ec581.png)

![image](https://user-images.githubusercontent.com/30885718/67658384-64047300-f994-11e9-8d0c-5b4feb79d0cf.png)

### What is expected?
import \_\_vue_normalize\_\_  from "rollup-plugin-runtime/runtime/normalize"

### What is actually happening?
import  \_\_vue_normalize\_\_ from 'C:Users
ongzongquanDesktopprojectue-components
ode_modules
ollup-plugin-vue
untime
ormalize.js';

![image](https://user-images.githubusercontent.com/30885718/67658381-623aaf80-f994-11e9-999c-a632096b3413.png)

**Comment by neacon-ivalkenburg** [comment]
Downgrading to rollup-plugin-vue latest 4.x fixed this for me.

**Comment by jameswragg** [comment]
I was also getting the following in my built files:
`import __vue_normalize__ from '../node_modules/rollup-plugin-vue/runtime/normalize.js';`

I experimented with downgrading to 5.1.1, which fixed for me:
`npm i rollup-plugin-vue@5.1.1 --save-dev`

**Comment by justin-schroeder** [comment]
@jameswragg is right, 5.1.2 introduced a dependency on `vue-runtime-helpers` in my `esm` and `umd` builds which I would prefer not to include, but maybe I just need to understand the issue better? Either way `5.1.2` appears to have a breaking change.

**Comment by hosein2398** [comment]
So can maintainers provide more information on this one?
What exactly is this?!
`import __vue_normalize__ from './node_modules/rollup-plugin-vue/runtime/normalize.js';`
So everyone who is using my bundle should install this lib to make bundle work?!
Or we should add this via rollup `external` or what?

**Comment by souse** [comment]
in window pc not work @5.1.2

**Comment by chaddugas** [comment]
This is still an issue - is there a fix coming for this anytime soon?

**Comment by znck** [comment]
I think the unexpected path in import statement is caused by `require.resolve`: https://github.com/vuejs/rollup-plugin-vue/blob/master/src/index.ts#L179. Can someone verify if changing `require.resolve` to `path.resovle(__dirname, '...')` to works on windows?

**Comment by znck** [comment]
@hosein2398 You can bundle or externalize it. To bundle include `rollup-plugin-node-resolve` in your config. To externalized add it as a dependency and include in rollup `external` plugin.

I suggest to externalize it.

**Comment by nicolas-t** [comment]
Hello, 

These failed :  
`opts.normalizer = '~' + path.resolve(__dirname, '../runtime/normalize');`
`opts.normalizer = '~' + path.join(__dirname, '../runtime/normalize');`

I suppose it's because of the `~`

Is `opts.normalizer` supposed to be a String, a Function or an Object ? 

If it's a String wouldn't the solution looks like this : 
`opts.normalizer = '"' + path.join(__dirname, '../runtime/normalize') + '"';`

**Comment by taoabc** [comment]
same problem

**Comment by VeraDidenko** [comment]
The problem occurs on Windows. On Ubuntu 18.04 everything works as expected. 
I hope this helps to find a solution.

**Comment by fubangfu** [comment]
vuePlugin( {
      needMap       : false,
      normalizer    : 'Vue.__normalizer',
      styleInjector : 'Vue.__styleInjector'
    } )
我目前是这样处理的。不过这两个方法，就得另外打包

**Comment by gslama-akqa** [comment]
Hi everyone,

I was also experiencing this problem and adding `rollup-plugin-node-resolve` solved it.

My configuration is now :

```
import resolve from 'rollup-plugin-node-resolve'
import commonjs from 'rollup-plugin-commonjs'
import VuePlugin from 'rollup-plugin-vue'
import pkg from './package.json'

export default {
  input: 'src/main.js',
  output: {
    file: pkg.main,
    format: 'cjs'
  },
  plugins: [
    resolve(),
    commonjs(),
    VuePlugin()
  ]
}
```

I wish I could explain why though...

**Comment by ProticM** [comment]
Same problem here.

@gslama-akqa adding `rollup-plugin-node-resolve` didn't solve it for me. Downgrading the versions did.

**Comment by gslama-akqa** [comment]
Hi @ProticM,

Not sure why this is working for me.

In case it can help with investigation, I pushed a working demo [here](https://github.com/gslama-akqa/rollup-demo).

**Comment by MrDrummer** [comment]
Update: Forced using 5.1.1 and it seems to have worked. I just didn't fully understand how yarn worked so misread the lock file, meaning everything was using the latest 5.1.4

**Comment by odai-alali** [comment]
with `rollup-plugin-vue@5.1.4`, I am getting the same problem when I build my package using `npm`. Using `yarn` solved the problem for me.

**Comment by houd1ni** [comment]
> 
> 
> with `rollup-plugin-vue@5.1.4`, I am getting the same problem when I build my package using `npm`. Using `yarn` solved the problem for me.

Super elegant solution, m8 🤣

**Comment by odai-alali** [comment]
> > with `rollup-plugin-vue@5.1.4`, I am getting the same problem when I build my package using `npm`. Using `yarn` solved the problem for me.
> 
> Super elegant solution, m8 🤣

Neither switching to yarn nor downgrading the package version are solutions for this problem. It's just a feedback about the problem that may help fixing it.

**Comment by mgdodge** [comment]
TL;DR - this should be rolled back and released as version 6.0.0 since it is a breaking change.

---

### Rationale

The real problem here is that this was released as a **semver-_patch_** change, when it probably should have been a **semver-_major_** change due to the fact that it is breaking - it forces on us a new dependency. I had to release a patch for my [cli utility](https://github.com/team-innovation/vue-sfc-rollup) locking `rollup-plugin-vue` to 5.1.1 because of this.

If I have a simple 'counter' demo component similar to the one in the [official vue docs](https://vuejs.org/v2/guide/components.html) - no external requirements at all, using it should not not introduce any dependencies beyond vue itself. So this is exactly what I did. When I built with 5.1.1 and compared it with that produced with 5.1.2+, the newer one complains about missing dependencies. **It is a breaking change** for me as a user of `rollup-plugin-vue` unless I now install `@rollup/plugin-node-resolve` and update my build process.

If I update my build process as suggested, and externalize the new `vue-runtime-helpers` dependency, this only passes the issue on to users of my components. **It is now a breaking change** for my own users! To be fair, I can choose to bundle the helpers, which fixes things and the output is exactly as it was before...but this was something that was only made possible by following this issue thread and waiting almost a month before the `@rollup/plugin-node-resolve` [comment](https://github.com/vuejs/rollup-plugin-vue/issues/303#issuecomment-557145401) was made.

I understand the desire to externalize `vue-runtime-helpers` to produce smaller bundles. But this whole issue is a prime example of how a project can benefit from semver - breaking changes should come with documentation of what is breaking, why, and how users of a package can update their projects.

This comment is starting to sound like a rant, which it wasn't meant to be, so let me wrap up.

I would suggest that this change (3b0b4de) be rolled back to restore original usage for those who expect the 5.1.1 configuration and output - it's not even clear what bug it was trying to address. Then, if truly necessary, release this change as 6.0.0 with the bug and new dependency/configuration requirements clearly documented. That way, additional users of `rollup-plugin-vue` aren't blindsided with this, and everyone who wishes to update has the choice to either explicitly bundle the helpers, or pass the new dependency on to their own users with their own **semver-major** change as appropriate.

**Comment by kling90** [comment]
Couldn't agree with @mgdodge more.
We also had to lock version 5.1.1 in our build environment while we were hoping this would be resolved.

**Comment by jeremycook** [comment]
Updated docs and cookbook would be a big help for me. I've tried going the recommended 5.1.4 route with externalized dependencies but can't get it to work. So sticking with 5.1.1 for now.

**Comment by znck** [comment]
Reverted to v5.1.1 source to release v5.1.5. 

> Will introduces new changes as BREAKING CHANGE.
